zerver/actions/message_edit.py

```diff
@@ -247,7 +247,14 @@ def get_mentions_for_message_updates(message_id: int) -> Set[int]:
             message=message_id,
             flags=~UserMessage.flags.historical,
         )
-        .filter(Q(flags__andnz=UserMessage.flags.mentioned | UserMessage.flags.wildcard_mentioned))
+        .filter(
+            Q(
+                flags__andnz=UserMessage.flags.mentioned
+                | UserMessage.flags.wildcard_mentioned
+                | UserMessage.flags.topic_wildcard_mentioned
+                | UserMessage.flags.group_mentioned
+            )
+        )
```

Comment:
This doesn't use the newly created index (wasn't using the earlier index too)

The sql we get is:
```sql
SELECT "zerver_usermessage"."user_profile_id" FROM "zerver_usermessage" WHERE 
("zerver_usermessage"."flags" | -1025 = -1025 AND "zerver_usermessage"."message_id" = 101 AND 
"zerver_usermessage"."flags" & 120 != 0)
```
The `flags=~UserMessage.flags.historical` filter being the reason.

Explain analyze result in dev:
```
 Index Scan using zerver_usermessage_message_id_f6e63a33 on zerver_usermessage  (cost=0.28..8.30 rows=1 width=4) (actual time=0
.034..0.039 rows=12 loops=1)
   Index Cond: (message_id = 101)
   Filter: (((flags & '120'::bigint) <> 0) AND ((flags | '-1025'::bigint) = '-1025'::integer))
 Planning Time: 3.225 ms
 Execution Time: 0.068 ms
(5 rows)
```

Reply:
I suspect this is because the statistics for the development environment are such that both flags are rare. You could try using `populate_db -n10000` to create more messages, sending a few dozen mentions, and then doing `CREATE STATISTICS` in postgres via a `manage.py dbshell` and see if that results in the index being used for this query.

Reply:
We can test the query, but based on the comment, the `historical` check is not actually necessary, just an "optimization", so we can just cut that if it's not actually faster.

Reply:
The limit on `message_id` is always going to dominate, and pick that index -- bitmap index scans are lossy, unlike normal index scans, so it's going to choose the `message_id` index preferentially.

But regardless, without that clause, it has no problems choosing the partial bitmap index, with an additional filter on `flags` for the historical bit.